### ISSUE TITLE
`SMFormingWall` is too complicated for most cases

### ISSUE THREAD

**Issue by pierreporte** [shaise/FreeCAD_SheetMetal#305]
*This issue has been thought of with the FreeCAD Design Working Group initiative for the sheet metal worbench conducted by @obelisk79 for @sliptonic. It is posted there because it is out of scope of this work.*

`SMFormingWall`  requires the user to make a different solid with the inverted shape they want to be stamped into the part. It allows a lot of freedom and doesn’t limit the user to a few simple shapes. However, most of the time, stamping is actually pretty simple: a flat bottom, a cutout, etc.

The workbench could provide a simplified version of `SMFormingWall` that takes a sketch as an input. The sketch would have been the outer shape of the solid. Potentially, the sketch could have several closed profiles. This new feature would provide some options to make a cutout (instead of selecting the opposite file) and allow to select edges in the sketch to mark that there must be an opening there (to make bridges or louvres). `SMFormingWall` would be used for the most complex cases.

Below are some examples from Solidworks and Catia to show how it could look.

--------------

![image](https://github.com/shaise/FreeCAD_SheetMetal/assets/2029853/45534d65-ca16-460f-a2ac-7ce8f21ccd7b)

Catia surface stamp: http://catiadoc.free.fr/online/cfyugstamp_C2/cfyugstampsurface.htm  
![image](https://github.com/shaise/FreeCAD_SheetMetal/assets/2029853/7c04ffd2-c656-4287-95de-8e08f36a3c30)

Catia flanged cutout: http://catiadoc.free.fr/online/cfyugstamp_C2/cfyugstampflangedcutout.htm  
![image](https://github.com/shaise/FreeCAD_SheetMetal/assets/2029853/2a08b6fe-d971-4875-84d3-3a998c928cb4)

Catia flanged hole: http://catiadoc.free.fr/online/cfyugstamp_C2/cfyugstampflangedhole.htm  
![image](https://github.com/shaise/FreeCAD_SheetMetal/assets/2029853/ff067d06-f935-41a5-817d-02f2a53e3d9a)

**Comment by pierreporte** [comment]
@shaise What do you think of this?

**Comment by shaise** [comment]
Yes, totally agree the forming tool is very complicated. But the idea of designing a solid with the desired punch shape seems to me more intuitive. The only problem is how to simplify the UI to use it.
A good task dialog can be a solution for that. Right now only the bend tool and base shape have usable task panels. I want to add better task panels for the other tools.

**Comment by pierreporte** [comment]
Designing a solid is not bad but it should be used only for the most complex cases. Even with a good UI, you’ll have some extra steps that are not necessary.

~~@FreeCAD/design-working-group can we propose an easy to use UI for this?~~ (ping doesn’t work outside the FreeCAD organization)

The most simple way to design a tool would be some sort of task panel that asks for relevant parameters only and generates the solid itself. It’s not that different that asking for an existing sketch and a few extra parameters like depth and radii as illustrated above. There could be an option to either generate a custom solid based on input, or provide an existing solid.

**Comment by shaise** [comment]
The good thing about using a sketch is that it does not involve using solids from other bodies (which tend to complicate part design, not only for sheetmetal) So I'm definitely in favor of adding this option as well. Coding-wise its quite complicated.

**Comment by pierreporte** [comment]
@sheetmetalman What about this?

**Comment by shaise** [comment]
@pierreporte , I'm already working on a task panel for that.

**Comment by pierreporte** [comment]
@shaise Great!

**Comment by sheetmetalman** [comment]
@pierreporte Indeed it will be a great improvement for `SheetMetal_Forming`!
I understand that @shaise is already taking care of this!

As @shaise said and looking at that Catia manual, I think using a sketch could be a good thing for this tool. Together with taskpanel will be asy to use and understand. Better UI/UX!

Perhaps an initial idea could be that the user could select a solid or a sketch for reference.
If the user selects a solid, it would work as it currently does.
But if user select a sketch, the tool will first create an extrusion and then use this extrusion in the same way the tool uses the solid. Of course, that extrusion will not be an object in the model tree, it will only exist in code.

**Comment by pierreporte** [comment]
@shaise Nearly one year ago you said you were working on a task panel. Do you have any news?

**Comment by shaise** [comment]
@pierreporte ,

Yes! there is a task panel for that:

<img width="371" height="557" alt="Image" src="https://github.com/user-attachments/assets/68e82c55-3966-425b-a16c-cb45d4b56b02" />

**Comment by pierreporte** [comment]
Indeed! Though it is hard to discover because of #480.

**Comment by shaise** [comment]
Yes, #480 is yet to be implemented
